myia/ir/node.py

```diff
@@ -74,6 +74,25 @@ def apply(self, fn, *inputs):
 
         return Apply(self, *edges)
 
+    def clone(self):
+        """Make a copy of this graph."""
+        res = Graph(self.parent)
+        objmap = {self: res}
+        res.parameters = [p.clone(self, objmap) for p in self.parameters]
+        res.flags = self.flags.copy()
+        if self.return_:
+            res.return_ = self.return_.clone(self, objmap)
+        res.varargs = (
+            self.varargs.clone(self, objmap) if self.varargs else self.varargs
+        )
+        res.kwargs = (
+            self.kwargs.clone(self, objmap) if self.kwargs else self.kwargs
+        )
+        res.defaults = self.defaults
```

Comment:
Is there any chance of this dict being modified?

```suggestion
        res.defaults = dict(self.defaults)
```

Reply:
That dictionary should not be modified because it's the default values for the arguments.  The expression could be optimized, but that should be done once replaced in the call expression, not in  there.